repo_autoindex/_impl/kickstart.py

```diff
@@ -0,0 +1,235 @@
+
+        # Only attempt to parse a supported treeinfo format
+        if treeinfo["header"]["type"] == "productmd.treeinfo" and treeinfo["header"][
+            "version"
+        ] in ["1.2"]:
+            for image in treeinfo["checksums"]:
+                entry = IndexEntry(
+                    href=image,
+                    text=os.path.basename(image),
+                )
+                if entry.href.endswith(".iso") or entry.href.endswith(".img"):
+                    entry.icon = ICON_OPTICAL
+                out.append(entry)
+        return out
+
+    async def _extra_files_entries(self) -> list[IndexEntry]:
+        """
+        An extra_files.json file might look like this:
+
+        {
+            "data": [
+                {
+                    "checksums": {
+                        "md5": "feb4d252ee63634debea654b446e830b",
+                        "sha1": "a73fad5aeb5642d1b2108885010c4e7a547a1204",
+                        "sha256": "c4117d0e325cde392981626edbd1484c751f0216689a171e4b7547e8800acc21"
+                    },
+                    "file": "RPM-GPG-KEY-redhat-release",
+                    "size": 5134
+                },
+                {
+                    "checksums": {
+                        "md5": "3c24137e12ece142a27bbf825c256936",
+                        "sha1": "a72daf8585b41529269cdffcca3a0b3d4e2f21cd",
+                        "sha256": "3f8644b35db4197e7689d0a034bdef2039d92e330e6b22217abfa6b86a1fc0fa"
+                    },
+                    "file": "RPM-GPG-KEY-redhat-beta",
+                    "size": 1669
+                },
+                {
+                    "checksums": {
+                        "md5": "b234ee4d69f5fce4486a80fdaf4a4263",
+                        "sha1": "4cc77b90af91e615a64ae04893fdffa7939db84c",
+                        "sha256": "8177f97513213526df2cf6184d8ff986c675afb514d4e68a404010521b880643"
+                    },
+                    "file": "GPL",
+                    "size": 18092
+                },
+                {
+                    "checksums": {
+                        "md5": "0c53898068810a989fa59ca0656bdf24",
+                        "sha1": "42d51858642b8a0d10fdf09050266395544ea556",
+                        "sha256": "8f833ce3fbcbcb82e47687a890c043332c88350ddabd606201556e14aaf8fcd9"
+                    },
+                    "file": "EULA",
+                    "size": 8154
+                }
+            ],
+            "header": {
+                "version": "1.0"
+            }
+        }
+        """
+        out: list[IndexEntry] = [
+            IndexEntry(
+                href="extra_files.json",
+                text="extra_files.json",
+                size=str(len(self.extra_files_content)),
+            ),
+        ]
+
+        extra_files = json.loads(self.extra_files_content)
+        if extra_files.get("header"):
+            # Only attempt to parse a supported extra_files format
```

Comment:
Same thing here, I'm not really in favor of only using the data for a specific known version.  I'd rather not have the check, or check for `startswith("1.")`.

If a new version is going to be introduced I find it unlikely that we'll become aware of it and update this code in advance, so this seems like setting things up to fail.